### ISSUE TITLE
Download all Posts from a Saved Collection

### ISSUE THREAD

**Issue by xelarate86** [instaloader/instaloader#544]
**Provide us a use case of the feature**
`instaloader --login=your_username :saved:COLLECTIONNAME`

**Describe the solution you'd like**
Instead of blanket downloading all saved items, could we create way to download only one collection? This would make it easier to organize saved files.

**Describe alternatives you've considered**
N/A

**If the feature request is accepted, would you be willing to submit a PR?**
Unfortunately I don't have the wherewithal to program this atm

**Additional context**
Reopening #166 2yrs later

(please also see https://instaloader.github.io/contributing.html for how to suggest a feature)

**Comment by aandergr** [comment]
This would be a great feature and we would accept a PR for that! I also think that `:saved:COLLECTIONNAME` would be okay as target name. However, since the Saved Collections are not available in the official Instagram Web App, it might be a little harder than usual to figure out how to obtain the required information.

**Comment by weiweihuanghuang** [comment]
I'm looking for this feature too

**Comment by federicoromeo** [comment]
I think that now the Instagram Web App provides the distinction between the different saved elements' *folders*. There will be an upgrade that provides the saved posts belonging only to a chosen folder? (or at least provide the name of the folder when retrieving the posts with Profile.get_saved_posts()

**Comment by xelarate86** [comment]
Looks pretty simple on their end, too. The URL structure is:
`https://www.instagram.com/USERNAME/saved/ALBUMNAME`

**Comment by federicoromeo** [comment]
> Looks pretty simple on their end, too. The URL structure is:
> `https://www.instagram.com/USERNAME/saved/ALBUMNAME`

But with this library it's not possible to do so yet, am i right?

**Comment by skoshy** [comment]
This feature would be really useful, could it be considered again?

**Comment by NickPfeiffer** [comment]
This would really be a helpful feature!

**Comment by federicoromeo** [comment]
I agree, i'd love it!

**Comment by policetonyR** [comment]
Could this be brought back up again?

**Comment by weiweihuanghuang** [comment]
Bump

**Comment by orazdow** [comment]
Why did they close this?

**Comment by federicoromeo** [comment]
It'd be super useful!
It would be even better if the posts could be accessed as streams.

**Comment by GamerGirlandCo** [comment]
looking at the saved JSON data of downloaded instagram posts reveals that they have a `saved_collection_ids` field, which is an array that contains ids of all the collections a post has been saved to.
there is also an api endpoint where you can fetch a user's collections (including the ids)
`https://i.instagram.com/api/v1/collections/list`.

**Comment by federicoromeo** [comment]
So i cannot directly retrieve all posts from a single collection from its ID, i can just retrieve all saved posts and see which ID(s) it belongs to, right?

**Comment by GamerGirlandCo** [comment]
> So i cannot directly retrieve all posts from a single collection from its ID, i can just retrieve all saved posts and see which ID(s) it belongs to, right?

yep!

**Comment by Cyph3r** [comment]
Weird reply. Issue #1172 is a much better answer.

**Comment by emsieler** [comment]
+1, would love this feature !

**Comment by prmichaelsen** [comment]
I'm happy to take a crack at this is a PR is welcome

**Comment by federicoromeo** [comment]
I can provide a python script for that

**Comment by aandergr** [comment]
> I'm happy to take a crack at this is a PR is welcome

Absolutely, a PR would be very welcome and we will be happy to review it!

**Comment by chanmathew** [comment]
Hi everyone, has there been any update on this? And does the download work with collections/saved folders that are shared with your account, and not owned by you?

**Comment by prmichaelsen** [comment]
Reporting back that I did not manage to get around to this. There's something wrong with my python3.9 install and I'm not very familiar with python, so I'm going to take a bow here.

When trying to run instaloader, I get
```
urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'OpenSSL 1.0.2k-fips  26 Jan 2017'. See: https://github.com/urllib3/urllib3/issues/2168
```

Note I followed [this guide](https://github.com/urllib3/urllib3/issues/3016) to update my install for openssl with no luck. Hopefully someone else is able to take a look at this

**Comment by cosmic-fire-eng** [comment]
+1 on this — my use case is archiving a few themed collections without pulling all of :saved every time. One data point for whoever picks it up: gallery-dl already scopes downloads to a single named collection (its InstagramCollectionExtractor handles URLs like instagram.com/<user>/saved/<name>/<id>), so the private API definitely supports fetching one collection's post list. If nobody else gets to it, I may take a crack at a PR. Thanks!

**Comment by Karthik-s10** [comment]
I'm happy it's reopened, this will be so useful
